internal/pkg/gateway/apiutils.go

```diff
@@ -78,6 +78,9 @@ func getValueFromResponse(response *peer.ProposalResponse) (*gateway.Result, err
 }
 
 func getChannelAndChaincodeFromSignedProposal(signedProposal *peer.SignedProposal) (string, string, error) {
+	if signedProposal == nil {
```

Comment:
I think this should find a home in protoutils/proputils.go (proposal utils). Some of the helpers over there do a fair bit of this and it would make sense to keep this sort of goo in that package since we already have a dependency on it.